### PR TITLE
feat(data): add concatenated name

### LINKED ISSUE #33: add alternative name without dashes
for `create-react-native-app` this should be `createreactnativeapp`. This is because it's often easier to just write it without dashes.

To remove is:

* `.`
* `-`
* `_`
* `@`
* `/`

The alternative name should have a lower relevancy score than the real name to avoid `react-native` to score above `reactnative` when you search for `reactnative`.

This should also be done for the keywords, but in that case the concatenated version can simply be added.

### PR DESCRIPTION
**Pull request by Haroenv**
fixes #33

### CODE REVIEW

File: formatPkg.js [hunk 1]
```diff
@@ -58,11 +58,17 @@ export default function formatPkg(pkg) {
   }
 
   const dependencies = cleaned.dependencies || {};
+  const charsToSplit = /[-/@_.]+/;
```
Comment (Haroenv) [inline formatPkg.js]: added `.` to split on here too

File: formatPkg.js [hunk 2]
```diff
@@ -58,11 +58,17 @@ export default function formatPkg(pkg) {
...
+    new RegExp(charsToSplit.source, 'g'),
+    '',
+  );
 
   const rawPkg = {
     objectID: cleaned.name,
     name: cleaned.name,
-    numberOfWordsInName: cleaned.name.split(/[-/@_]+/).length,
+    concatenatedName,
+    numberOfWordsInName: cleaned.name.split(charsToSplit).length,
```
Comment (iam4x) [inline formatPkg.js]:
```js
'r-a-b-c'.split(/[-/@_.]+/g)
-> (4) ["r", "a", "b", "c"]
```

Comment (Haroenv) [inline formatPkg.js]: thanks!

**Comment by Haroenv** [comment]
I removed `numberOfWordsInName` because it's not useful anymore when the concatenated name strategy is used.

**Pull request merged**